chapters/classes.tex

```diff
@@ -341,21 +341,21 @@ \subsection{Component Variability Prefixes discrete, parameter, constant}\label{
 \begin{itemize}
 \item
   A variable \lstinline!vc! declared with \lstinline!constant!\indexinline{constant} prefix remains constant during transient analysis, with a value that is unaffected by the initialization problem.
+  This is called a \firstuse{constant}, or \firstuse{constant variable}\index{constant variable}\index{component variability!constant}.
 \item
   A variable \lstinline!vc! declared with the \lstinline!parameter!\indexinline{parameter} prefix remains constant during transient analysis, with a value determined by the initialization problem.
+  This is called a \firstuse{parameter}, or \firstuse{parameter variable}\index{parameter variable}\index{component variability!parameter}.
 \item
-  A \emph{discrete-time} variable \lstinline!vd! has a vanishing time derivative between events.
-  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere,
-  as the derivative is not even defined at the events, and it is not legal
-  to apply \lstinline!der! to discrete-time variables as they are not continuous. During transient analysis the variable
-  can only change its value at event
-  instants (see \cref{events-and-synchronization}).
+  A \emph{discrete-time} variable \lstinline!vd! is a variable that is discrete-valued (that is, not of \lstinline!Real! type) or assigned in a \lstinline!when!-clause.
+  The \lstinline!discrete!\indexinline{discrete} prefix may be used to clarify that a variable is discrete-time.
+  It has a vanishing time derivative between events.
+  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere, as the derivative is not even defined at the events.
+  It is not allowed to apply \lstinline!der! to discrete-time variables.
+  During transient analysis the variable can only change its value at event instants (see \cref{events-and-synchronization}).
 \item
-  A \emph{continuous-time} variable \lstinline!vn! may have a non-vanishing time
-  derivative (\lstinline!der(vn)<>0! possible) and may also
-  change its value discontinuously at any time during transient analysis
-  (see \cref{events-and-synchronization}). If there are any discontinuities the variable is
-  not differentiable.
+  A \emph{continuous-time} variable is a \lstinline!Real! variable without any prefix that is not assigned in a \lstinline!when!-clause.
+  A continuous-time variable \lstinline!vn! may have a non-vanishing time derivative (\lstinline!der(vn)<>0! possible) and may also change its value discontinuously at any time during transient analysis (see \cref{events-and-synchronization}).
```

Comment:
This comment belongs to line 356: Introduce _continuous-time variable_ with `\firstuse` and `\index`.

Reply:
will add index, but unfortunately first-use is not versatile enough for this.

Reply:
What's the problem with `\firstuse{continuous-time variable}`?

Reply:
The example below uses `\index{continuous time!variable}`; so if we later want `\firstuse` to map to an index-entry we would need to refine it, I thought it was easier to just use `\emph` for the time being.
There's also, separately, a need for cleaning up the index.

Reply:
Looks good now.  Resolving.